### PR TITLE
kratuwu

### PR DESCRIPTION
**Pull request by andyngdz**

### CODE REVIEW

File: challenge-react/src/App.js [hunk 1]
```diff
@@ -1,111 +1,63 @@
...
-  border: 1px solid #ccc;
+import { summaryDonations } from './helpers';
+import AppHeader from './components/AppHeader';
+import DonateList from './components/DonateList';
+import charityApi from './api/charityApi';
+import paymentApi from './api/paymentApi';
+import { useDonate } from './context/donateContext';
+import Popup from './components/Popup';
+import styled from 'styled-components';
+const Container = styled.div`
```
Comment (EpicHigh) [inline challenge-react/src/App.js:11]: You separate the other components, but this one does not.

File: challenge-react/src/App.js [hunk 2]
```diff
@@ -1,111 +1,63 @@
...
+const App = () => {
+  const donateContext = useDonate();
+  const [charities, setCharities] = useState([]);
+  const [showPopup, setShowPopup] = useState(false);
+  const [popupMessage, setPopupMessage] = useState();
+  useEffect(() => {
+    charityApi.getCharities().then((data) => {
+      setCharities(data);
+    });
+    updateTotalPayment();
```
Comment (EpicHigh) [inline challenge-react/src/App.js:23]: This function was used before it was defined.

File: challenge-react/src/App.js [hunk 3]
```diff
@@ -1,111 +1,63 @@
...
+  }, []);
+
+  const handlePay = (charitiesId, amount, currency) => {
+    paymentApi
+      .updatePayment({ charitiesId, amount, currency })
+      .then(() => {
+        updateTotalPayment();
```
Comment (EpicHigh) [inline challenge-react/src/App.js:30]: This function was used before it was defined.

File: challenge-react/src/components/DonateCard.jsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  display: flex;
+  border-radius: 3px;
+  position: relative;
+  box-shadow: 2px 2px 4px 0px #e2e2e2;
+  flex-direction: column;
+`;
+
+const CartMedia = styled.img`
+  height: calc(100% - 70px);
+  height: 100%;
```
Comment (EpicHigh) [inline challenge-react/src/components/DonateCard.jsx:20]: Property height is overwritten .

File: challenge-react/src/components/DonateForm.jsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+const DonateForm = (props) => {
+  return (
+    <FormWrapper>
+      <h3>Select the amiunt to donate ({props.currency})</h3>
+      <FormStyled onSubmit={props.onSubmit}>
+        <div>
+          {[10, 20, 50, 100, 500].map((amount, j) => (
+            <RadioWrapper key={amount}>
+              {j === 0 ? (
+                <input type="radio" name="payment" value={amount} required />
```
Comment (EpicHigh) [inline challenge-react/src/components/DonateForm.jsx:35]: `<input type="radio" name="payment" value={amount} required={j === 0} />`

File: challenge-react/src/components/DonateForm.jsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+`;
+
+const RadioWrapper = styled.label`
+  margin-right: 5px;
+`;
+
+const DonateForm = (props) => {
+  return (
+    <FormWrapper>
+      <h3>Select the amiunt to donate ({props.currency})</h3>
```
Comment (EpicHigh) [inline challenge-react/src/components/DonateForm.jsx:29]: Typo: amiunt

File: challenge-react/src/components/Popup.jsx [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+const Overlay = styled.div`
+  position: fixed;
+  top: 0;
+  bottom: 0;
+  left: 0;
+  right: 0;
+  background: rgba(0, 0, 0, 0.7);
+  transition: opacity 500ms;
+  visibility: hidden;
```
Comment (EpicHigh) [inline challenge-react/src/components/Popup.jsx:13]: Property visibility is overwritten

File: challenge-react/src/helpers.js [hunk 8]
```diff
@@ -1,2 +1,2 @@
 export const summaryDonations = (danations) =>
-  danations.reduce((accumulator, value) => accumulator + value);
+  danations.reduce((accumulator, value) => accumulator + (value || 0));
```
Comment (EpicHigh) [inline challenge-react/src/helpers.js:2]: `danations.reduce((accumulator, value) => accumulator + value, 0));`

**Review by EpicHigh — COMMENTED** [review]
Recommendation

I do not prefer snapshot testings because our web-page is frequently changing based on requirements. They are very fragile from changes. It makes us manually verify that everything is still working properly and update the snapshots again. The solution is we write the tests that make sure the components are working for users instead.

File: challenge-react/src/components/DonateCard.jsx [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+  flex-grow: 1;
+  margin: 20px;
+  align-items: center;
+`;
+
+const DonateCard = (props) => {
+  const [formVisible, setFormVisible] = useState(false);
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    props.onSubmit(
```
Comment (andyngdz) [inline challenge-react/src/components/DonateCard.jsx:38]: Looks dirty here could it be:
const { onSubmit , item } = props;

```
const handleSubmit = (e) => { 
    e.preventDefault();
    onSubmit(item...);
}
```

File: challenge-react/src/components/DonateCard.jsx [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+    toggleDonateForm();
+  };
+
+  const toggleDonateForm = () => {
+    setFormVisible(!formVisible);
+  };
+  return (
+    <Card className={props.className}>
+      <Modal visible={formVisible} onClose={toggleDonateForm}>
+        <DonateForm onSubmit={handleSubmit} currency={props.item.currency} />
```
Comment (andyngdz) [inline challenge-react/src/components/DonateCard.jsx:53]: Can reuse the item from destruction method above.

**Review by andyngdz — COMMENTED** [review]
Good job

**Pull request closed without merging**